src/runtime/template_set.ts

```diff
@@ -97,7 +108,8 @@ export class TemplateSet {
       this.templates[name] = function (context, parent) {
         return templates[name].call(this, context, parent);
       };
-      const template = templateFn(this, bdom, helpers);
+      const runtimeUtils = { ...helpers, __globals__: { ...this.globalValues } };
```

Comment:
not sure why you spread this.globalValues? is there a need to specifically do that to avoid modifying it? 
```js
      const runtimeUtils = { ...helpers, __globals__: this.globalValues };
```